### PR TITLE
Feature/approve match

### PR DESCRIPTION
**Pull request by vinutama**

### CODE REVIEW

File: repository/match/match_repository_impl.go [hunk 1]
```diff
@@ -16,36 +19,101 @@ func NewMatchRepository() MatchRepository {
...
+		return exc.InternalServerException(fmt.Sprintf("Internal server error when get remain match issuer: %s", err))
+	}
+	remainMatchCatReceiverIds, err := getRemainingMatchCat(ctx, tx, catReceiverId)
+	if err != nil {
+		return exc.InternalServerException(fmt.Sprintf("Internal server error when get remain match receiver: %s", err))
+	}
+	remainMatchIds := append(remainMatchCatIssuerIds, remainMatchCatReceiverIds...)
+
+	// delete if any remain match requested on both cats
+	if len(remainMatchIds) > 0 {
```
Comment (kangman53) [inline repository/match/match_repository_impl.go]: bwang ini deletenya kenapa nggak dijadiin 1 query aja dah?
pake or gitu? catIssuerId = catIssuerId OR catIssuerId = catReceiverId OR catReceiverId = catIssuerId OR catReceiverId = catReceiverId

Comment (vinutama) [reply repository/match/match_repository_impl.go]: eh iye maksudnyaa pas getRemainCatsnya yak?

**Comment by rizqitaufiqf** [comment]
kalo dr aku udah gg mas

File: repository/match/match_repository_impl.go (hunk 1 above)
Comment (kangman53) [reply repository/match/match_repository_impl.go]: ah nvm pak harusnya itu nggak perlu getRemainCats kan? langsung lu deletenya pake OR itu. wkwkkwkwk

Comment (vinutama) [reply repository/match/match_repository_impl.go]: wkwkw bener juga lu ngapa ribet2 ya gw. okeoke pakk

**Pull request merged**